### PR TITLE
ENG-1232: add support for enum16 and add ch enum validation

### PR DESCRIPTION
**Pull request by LucioFranco**
> [!NOTE]
> Adds full signed ClickHouse Enum8/Enum16 support (including negative values) across parsing, codegen, and validation, with comprehensive tests and documentation.
> 
> - **Core Types**:
>   - Change enum integer representation from `u8` to `i16` in `EnumValue`/metadata and TS generator (`TSEnumValue.Number(i16)`).
>   - Safe proto conversion for enum ints with range checks; add unit tests for valid/out-of-range cases.
> - **ClickHouse Parsing/Conversion**:
>   - Tokenizer now supports signed `NumberLiteral`; Enum regex supports negatives in `infrastructure/olap/clickhouse/model.rs`.
>   - Validate Enum8/Enum16 ranges and convert to `Enum<Int(i16)>` in `type_parser.rs`.
>   - Reject negative JSON params for `max_dynamic_types`/`max_dynamic_paths`.
>   - Add extensive tests for Enum8/16 boundaries, negatives, out-of-range errors, and JSON param validation.
> - **Python Parser**:
>   - Parse integer enum members as `i16` with clear out-of-range errors.
> - **Validation Runtime (`validate_passthrough.rs`)**:
>   - Adjust enum integer conversions (`from_i16`) and prevent negative enum values from matching unsigned (`u64`) inputs.
>   - Add tests covering negative enum handling for `u64`/`i64` and map key validation.
> - **E2E Tests (db-pull)**:
>   - Add Enum16 column with large values; verify generated Python/TS code preserves enum values and default syntax.
> - **Docs**:
>   - Document integer-backed enums with examples and Enum8/Enum16 ranges in `supported-types.mdx`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ccef56b263bbc9482f1e690ed2e1192eae9c64df. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/framework-cli/src/framework/core/infrastructure/table.rs [hunk 1]
```diff
@@ -172,8 +172,8 @@ pub struct EnumMemberMetadata {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 #[serde(untagged)]
 pub enum EnumValueMetadata {
-    /// Integer value for numeric enums
-    Int(u8),
+    /// Integer value for numeric enums (supports Enum8: -128 to 127, Enum16: -32768 to 32767)
+    Int(i16),
```
Comment (LucioFranco) [inline apps/framework-cli/src/framework/core/infrastructure/table.rs:176]: This kinda sketches me out but feels safe since u8 is a subset of i16 anyways. Though this kinda forces us to follow the clickhouse thing of supporting negative enum ids.

**Review by phiSgr — APPROVED** [review]
do we want to allow the user to explicitly specify enum16 even if the numbers are in range for 8-bit?

if so we can add handling for `ClickHouseByteSize`

**Comment by LucioFranco** [comment]
@phiSgr thats a really good idea, extracting it to an issue now so I can do this in a follow up.
